gradio/processing_utils.py

```diff
@@ -462,6 +457,55 @@ def _move_to_cache(d: dict):
     return client_utils.traverse(data, _move_to_cache, client_utils.is_file_obj)
 
 
+def _check_allowed(path: str | Path, check_in_upload_folder: bool):
+    blocks = LocalContext.blocks.get()
```

Comment:
is not sufficient to use Context.root_block?

Reply:
I think we set Context.root_block to `None` when we exit the Blocks context and launch

Reply:
Ah I see